plugins/ufm_log_analyzer_plugin/src/loganalyze/log_analyzers/base_analyzer.py

```diff
@@ -29,17 +29,96 @@
 warnings.filterwarnings("ignore")
 
 
-class BaseAnalyzer:
-    """
-    Analyzer class that gives all the logs the
-    ability to print/save images and filter data
-    """
-
+class BaseImageCreator:
     # Setting the graph time interval to 1 hour
     # This is out side of the constructor since
     # It is defined once for all graphic options
     # Per call/run.
+
     time_interval = "h"
+    _images_type = {"svg", "png"}
+    def __init__(self, dest_image_path):
+        self._dest_image_path = dest_image_path
+        self._images_created = []
+        self._funcs_for_analysis = set()
+
+    def _plot_and_save_data_based_on_timestamp(
+        self, data_to_plot, x_label, y_label, title
+    ):
+        with plt.ion():
+            log.LOGGER.debug(f"saving {title}")
+            plt.figure(figsize=(12, 6))
+            plt.plot(data_to_plot, marker="o", linestyle="-", color="b")
+            plt.title(title)
+            plt.xlabel(x_label)
+            plt.ylabel(y_label)
+            plt.grid(True)
+
+            # Set the locator to show ticks every hour and the formatter to
+            # include both date and time
+            ax = plt.gca()
+            ax.xaxis.set_major_locator(mdates.HourLocator())
+            ax.xaxis.set_minor_locator(
+                mdates.MinuteLocator(interval=15)
+            )  # Add minor ticks every 15 minutes
+            ax.xaxis.set_major_formatter(
+                mdates.DateFormatter("%Y-%m-%d %H:%M")
+            )  # Format major ticks to show date and time
+
+            plt.xticks(rotation=45)  # Rotate x-axis labels to make them readable
+            plt.tight_layout()
+
+            generic_file_name = f"{title}".replace(" ", "_").replace("/", "_")
+            images_created = []
+            for img_type in self._images_type:
+                cur_img = os.path.join(self._dest_image_path,f"{generic_file_name}.{img_type}")
+                log.LOGGER.debug(f"Saving {cur_img}")
+                plt.savefig(cur_img, format=img_type)
+                images_created.append(cur_img)
+            images_list_with_title = [(image, title) for image in images_created]
+            self._images_created.extend(images_list_with_title)
+            plt.close()
+
+    def _plot_and_save_pivot_data_in_bars(  # pylint: disable=# pylint: disable=too-many-arguments
```

Comment:
Yes, they have
I tried to move the plot logic to be common but I was not able to do it work since it is a module with his inner state.
Something did not work once I moved it to another function.